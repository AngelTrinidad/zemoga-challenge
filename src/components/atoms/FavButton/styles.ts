import {StyleSheet} from 'react-native';

import {scaleSpace} from '~helpers';

const styles = StyleSheet.create({
  image: {
    padding: scaleSpace(12),
    width: scaleSpace(20),
    height: scaleSpace(20),
  },
});

export default styles;
