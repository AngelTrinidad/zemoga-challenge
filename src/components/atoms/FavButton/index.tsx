import React, {FC} from 'react';
import {Image, TouchableOpacity, TouchableOpacityProps} from 'react-native';

import styles from './styles';

import FavPNG from '~assets/img/fav.png';
import UnfavPNG from '~assets/img/unfav.png';

interface Props extends TouchableOpacityProps {
  isFavorite: boolean;
}

const FavButton: FC<Props> = ({isFavorite, ...props}) => {
  return (
    <TouchableOpacity {...props}>
      <Image source={isFavorite ? FavPNG : UnfavPNG} style={styles.image} />
    </TouchableOpacity>
  );
};

export default FavButton;
