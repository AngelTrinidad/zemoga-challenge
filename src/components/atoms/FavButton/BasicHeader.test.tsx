import React from 'react';

import FavButton from '.';

import {render} from '~helpers/testing';

describe('<FavButton />', () => {
  it('should render correctly', () => {
    const component = render(<FavButton isFavorite />);
    expect(component).toBeDefined();
  });

  it('should has same snapshot(with isFavorite prop)', () => {
    const component = render(<FavButton isFavorite />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should has same snapshot(with isFavorite=false prop)', () => {
    const component = render(<FavButton isFavorite={false} />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
