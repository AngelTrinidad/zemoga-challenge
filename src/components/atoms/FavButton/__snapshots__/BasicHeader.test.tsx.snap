// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FavButton /> should has same snapshot(with isFavorite prop) 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <Image
    source={
      {
        "testUri": "../../../src/assets/img/fav.png",
      }
    }
    style={
      {
        "height": 22.857142857142858,
        "padding": 13.714285714285715,
        "width": 22.857142857142858,
      }
    }
  />
</View>
`;

exports[`<FavButton /> should has same snapshot(with isFavorite=false prop) 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <Image
    source={
      {
        "testUri": "../../../src/assets/img/unfav.png",
      }
    }
    style={
      {
        "height": 22.857142857142858,
        "padding": 13.714285714285715,
        "width": 22.857142857142858,
      }
    }
  />
</View>
`;
